mountpoint-s3-client/src/mock_client.rs

```diff
@@ -71,7 +71,7 @@ pub struct MockClientConfig {
 
 /// A mock implementation of an object client that we can manually add objects to, and then query
 /// via the [ObjectClient] APIs.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Something like this?

```rust
#[derive(Debug, Clone)]
pub struct MockClient {
    inner: Arc<MockClientInner>,
}

#[derive(Debug)]
pub struct MockClientInner {
    config: MockClientConfig,
    objects: Arc<RwLock<BTreeMap<String, MockObject>>>,
    in_progress_uploads: Arc<RwLock<BTreeSet<String>>>,
    operation_counts: Arc<RwLock<HashMap<Operation, u64>>>,
}
```

This will require a bunch of `self.<field>` to `self.inner.<field>` changes in the file, but looks ok. Cloning the struct field-by-fields just requires less changes, but is slightly less memory efficient, yes.

Reply:
We could likely review/remove the `Arc` in the fields though. No need to change now, I was only asking where you needed it `Clone`.

Reply:
Ah, ok! The clone is needed in [multilevel cache unit-tests](https://github.com/awslabs/mountpoint-s3/pull/1064/files/8d9efd66ec6264356979f290b8450c1d8860168c#diff-9106dd30662c843c8114dbe0453395ba43b722d8777fda494f020e820e8b1644R129-R131).